### PR TITLE
Dashboard JSON refactored

### LINKED ISSUE #224: add program information to dashboard API
restructure json returned from the dashboard api to include program information: return a nested structure like:

```
[
    {
        "id": 1,
        "title": "Program in economics", 
        "courses": [
            {
                "id": 99,
                "title": "Gio test course",
                "position_in_program" : 0,
                "runs": [
                    {
                        "course_id": "course-v1:Org+GIO101+CR-SPRING17",
                        "id": 10,
                        "status": "offered-not-enrolled",
                        "title": "Gio Test Course #2",
                        "enrollment_start_date": "2017-07-01T00:00:00Z",
                        "fuzzy_enrollment_start_date": ""
                    },
                    {
                        "course_id": "course-v1:Org+GIO101+CR-SPRING16",
                        "id": 9,
                        "status": "not-passed",
                        "title": "Gio Test Course #1",
                        "enrollment_start_date": "2016-07-01T00:00:00Z",
                        "fuzzy_enrollment_start_date": ""
                    }
                ]
            },
            { ... // another course with the runs }
        ]
    },
    { ... // another program with courses }
]
```

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
fixes #224

#### What's this PR do?
it refactors the structure of the JSON for the dashboard. 

#### Where should the reviewer start?
dashboard/api.py

#### How should this be manually tested?
The rest api endpoint for the dashboard should now produce a different output

#### Any background context you want to provide?
the dashboard UI will be broken after this until the UI part is done

### CODE REVIEW

File: dashboard/api.py [hunk 1]
```diff
@@ -123,47 +163,57 @@ def get_info_for_course(user, course, user_enrollments, user_certificates):
...
-        )
+        next_run = course.get_next_run()
+        status = CourseStatus.OFFERED if next_run is not None else CourseStatus.NOT_OFFERED
+        course_data['status'] = status
+        if next_run is not None:
+            course_data['runs'].append(format_course_for_dashboard(next_run, status))
     elif run_status.status == CourseRunStatus.GRADE:
-        return format_course_for_dashboard(run_status.course_run, CourseStatus.CURRENT_GRADE, course)
+        course_data['status'] = CourseStatus.CURRENT_GRADE
+        course_data['runs'].append(format_course_for_dashboard(run_status.course_run, CourseStatus.CURRENT_GRADE))
```
Comment (noisecapella) [inline dashboard/api.py]: Can you rename `format_course_for_dashboard` to `format_courserun_for_dashboard`?

Comment (giocalitri) [inline dashboard/api.py]: you are right: now the name is wrong

File: dashboard/views.py [hunk 2]
```diff
@@ -46,13 +46,7 @@ def get(self, request, *args, **kargs):  # pylint: disable=unused-argument, no-s
...
-        response_data = {'courses': []}
+        response_data = []
         for program in Program.objects.all():
-            for course in program.course_set.all():
-                response_data['courses'].append(
-                    get_info_for_course(
-                        request.user, course, enrollments, user_certificates=certificates)
-                )
-
-        response_data['courses'].sort(key=lambda x: x['position_in_program'])
```
Comment (noisecapella) [inline dashboard/views.py]: Do we care about the order of programs being returned?

**Comment by noisecapella** [comment]
Looks good to me :+1:

File: dashboard/views.py (hunk 2 above)
Comment (giocalitri) [inline dashboard/views.py]: I asked this question yesterday to @pdpinch and he replied something like "probably, but let's postpone this"

**Pull request merged**